src/m365/spo/commands/file/file-version-list.ts

```diff
@@ -0,0 +1,107 @@
+import { AxiosRequestConfig } from 'axios';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { formatting } from '../../../../utils/formatting';
+import { validation } from '../../../../utils/validation';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  fileUrl?: string;
+  fileId?: string;
+}
+
+class SpoFileVersionListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FILE_VERSION_LIST;
+  }
+
+  public get description(): string {
+    return 'List all the versions of a file in a SharePoint Document library';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        fileUrl: typeof args.options.fileUrl !== 'undefined',
+        fileId: typeof args.options.fileId !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-w, --webUrl <webUrl>'
+      },
+      {
+        option: '-u, --fileUrl [fileUrl]'
+      },
+      {
+        option: '-i, --fileId [fileId]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.fileId && !validation.isValidGuid(args.options.fileId)) {
+          return `${args.options.fileId} is not a valid GUID`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(['fileUrl', 'fileId']);
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving all versions of file ${args.options.fileUrl || args.options.fileId} at site ${args.options.webUrl}...`);
+    }
+
+    try {
+      let requestUrl = `${args.options.webUrl}/_api/web`;
+      if (args.options.fileUrl) {
+        requestUrl = `${args.options.webUrl}/_api/web/GetFileByServerRelativeUrl('${formatting.encodeQueryParameter(args.options.fileUrl)}')/versions`;
+      }
+      else {
+        requestUrl = `${args.options.webUrl}/_api/web/GetFileById('${args.options.fileId}')/versions`;
+      }
+      const requestOptions: AxiosRequestConfig = {
+        url: requestUrl,
+        headers: {
+          'accept': 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+
+      const response = await request.get<{ value: any[] }>(requestOptions);
```

Comment:
With a single GET request we might miss versions if a file has over 5k items. However at this moment we don't really have a util that is capable of fetching paged SP objects.
Let's ensure this request has a top of 5k so we get as much versions as possible. Meanwhile I'll discuss internally what we should do to retrieve all versions (I'm guessing we should create an `odata.getAllItems` function for the SP API).